druid/src/main/java/org/apache/calcite/adapter/druid/DruidConnectionImpl.java

```diff
@@ -338,8 +338,40 @@ private void parseField(List<String> fieldNames, List<ColumnMetaData.Rep> fieldT
       break;
     case VALUE_STRING:
     default:
-      rowBuilder.set(i, parser.getText());
-      break;
+      String s = parser.getText();
+      if (type != null) {
+        switch (type) {
+        case LONG:
+        case PRIMITIVE_LONG:
+        case SHORT:
+        case PRIMITIVE_SHORT:
+        case INTEGER:
+        case PRIMITIVE_INT:
+          if (s.equals("Infinity") || s.equals("-Infinity") || s.equals("NaN")) {
+            throw new RuntimeException("/ by zero");
+          }
+        case FLOAT:
+        case PRIMITIVE_FLOAT:
+        case PRIMITIVE_DOUBLE:
+        case NUMBER:
+        case DOUBLE:
+          if (s.equals("Infinity")) {
+            rowBuilder.set(i, Double.POSITIVE_INFINITY);
+            break;
```

Comment:
The reason I put break in each branch is because when the String calcite received is not "Infinite", "-Inifinty' and "NaN", it is possible that the String is a String representing float number. This case will be dealt in the default branch. I can change it to a switch cases to keep this logic. Or is it actually all return of Float calcite type can only be String when return is one of "Infinity", "NaN" and "-Infinity"?